addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2908,6 +2908,10 @@ var SnippetsMenu = Widget.extend({
         exclude += `${exclude && ', '}.o_snippet_not_selectable`;
 
         let filterFunc = function () {
+            if (forDrop && this.closest("[data-oe-type=image]")) {
```

Comment:
@bvr-odoo This comment still to handle:

> What if this is a data-oe-type=image itself? It would not be a child of an image field but your checks would say it is.

I think that:
- if drop-in, your condition is right
- if drop-near, your should do `this.parentNode.closest(...)`, otherwise you are preventing dropping **near** any image t-field

Reply:
@qsm-odoo I made the changes